pkg/services/librarypanels/librarypanels.go

```diff
@@ -67,4 +67,21 @@ func (lps *LibraryPanelService) AddMigration(mg *migrator.Migrator) {
 
 	mg.AddMigration("create library_panel table v1", migrator.NewAddTableMigration(libraryPanelV1))
 	mg.AddMigration("add index library_panel org_id & folder_id & name", migrator.NewAddIndexMigration(libraryPanelV1, libraryPanelV1.Indices[0]))
+
+	libraryPanelDashboardV1 := migrator.Table{
+		Name: "library_panel_dashboard",
+		Columns: []*migrator.Column{
+			{Name: "id", Type: migrator.DB_BigInt, IsPrimaryKey: true, IsAutoIncrement: true},
+			{Name: "librarypanel_id", Type: migrator.DB_BigInt, Nullable: false},
+			{Name: "dashboard_id", Type: migrator.DB_BigInt, Nullable: false},
+			{Name: "created", Type: migrator.DB_DateTime, Nullable: false},
+			{Name: "created_by", Type: migrator.DB_BigInt, Nullable: false},
```

Comment:
Shouldn't created_by be marked as a foreign key?

Reply:
Why, I don't think I understand can you elaborate?

Reply:
Do you mean a foreign key to the `user` table?

Reply:
created_by refers to a user ID, so shouldn't it be a foreign key relation (FK pointing from library_panel_dashboard to the user table)?

Reply:
Come to think of it, shouldn't librarypanel_id and dashboard_id also be FKs?

Reply:
I agree that librarypanel_id and dashboard_id should be foreign keys. Do you know if we have an example of foreign keys usage in Grafana?

Reply:
But I think `created_by` doesn't make sense as we're using the signed in users id. If that is wrong, then something else is seriously wrong, don't you think?

Reply:
I don't know directly how to make foreign key migrations in Grafana since I haven't had the need yet, but I would search the code to look for examples. There ought to be something.

About `created_by`, what's the concrete argument for not making it a foreign key referring to the user table (I'm not sure off the top of my head what is the exact table's name)? I don't follow you here.

Reply:
My argument was strictly from an api usage scenario, but if someone adds rows in the database manually then of course it makes sense, I'll add it as well. And I'll also add foreign keys to `folder_id`, `org_id`, `updated_by` and `created_by` in the `library_panel` table as well. The challenge I've is that I can't seem to find any foreign key usages in Grafana.

Reply:
@marefr @papagian we should use foreign key constraints for the aforementioned fields here, right?

Reply:
@hugohaggmark I don't really see what the FK has to do with it being used from the API or not. When for example a user is removed from the database, it should be detected that any corresponding `created_by` references are stale.

Reply:
I agree with you @aknuds1, but I can't see it happening using xorm and I can't see any examples throughout our codebase. Do you think we can add this when we know how to in a follow up PR instead?

Reply:
We don't use foreign keys for historical and technical reasons. Not something we should add in this PR ass Hugo suggest, but potentially something to investigate on its own

Reply:
@marefr Thanks for clarifying that, I've been able to dig up a [comment](https://github.com/grafana/grafana/issues/3269#issuecomment-383328548) by @torkelo that sheds light on why foreign key constraints aren't used in Grafana. I agree that there's no point in asking for FKs in this PR considering they're not used in Grafana.